tests/unit/base/Plugin_unittest.cc

```diff
@@ -0,0 +1,114 @@
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Written (W) 2016 Sanuj Sharma
+ */
+
+#include <../tests/unit/base/MockBaseClass.h>
+#include <gtest/gtest.h>
+
+using namespace shogun;
+
+TEST(Plugin, load_close)
+{
+    auto library = loadLibrary("./libplugin.so");
+    auto manifest = library.manifest();
+    auto mock_class = manifest.classByName<MockBaseClass>("mock_class");
+    auto another_mock_class = manifest.classByName<MockBaseClass>("another_mock_class");
+    auto mock_class_obj = mock_class.instance();
+    auto another_mock_class_obj = another_mock_class.instance();
+
+    std::string mock_class_name = mock_class_obj->get_name();
+    std::string another_mock_class_name = another_mock_class_obj->get_name();
+
+    EXPECT_EQ(mock_class_obj->mockMethod(), 0);
+    EXPECT_EQ(another_mock_class_obj->mockMethod(), 0);
+    EXPECT_EQ(mock_class_name, "MockClass");
+    EXPECT_EQ(another_mock_class_name, "AnotherMockClass");
+}
+
+TEST(Plugin, load_non_existent_lib)
+{
+    EXPECT_THROW(loadLibrary("./librandomplugin.so"), ShogunException);
+}
+
+TEST(Plugin, load_non_existent_class)
+{
+    auto library = loadLibrary("./libplugin.so");
+    auto manifest = library.manifest();
+    EXPECT_THROW(manifest.classByName<MockBaseClass>("arbitrary_class"), ShogunException);
+}
+
+TEST(Plugin, load_lib_wo_manifest)
+{
+    auto library = loadLibrary("./libpluginwomanifest.so");
+    EXPECT_THROW(library.manifest(), ShogunException);
+}
+
+TEST(Plugin, library_copy_constructor)
+{
+    Library library("./libplugin.so");
+    Library copy_library(library);
+    EXPECT_EQ(copy_library, library);
+}
+
+TEST(Plugin, library_assignment_operator)
+{
+    Library library("./libplugin.so");
+    Library copy_library = library;
+    EXPECT_EQ(copy_library, library);
+}
+
+TEST(Plugin, library_manifest_accessor_name)
+{
+    std::string name = Library::getManifestAccessorName();
+    EXPECT_EQ(name, "shogunManifest");
+}
+
+TEST(Plugin, manifest)
+{
+    Manifest manifest("Mock library",
+    {
+        std::make_pair("mock_base_class", erase_type(
+            MetaClass<CSGObject>(erase_type(
+                std::function<Some<CSGObject>()>(
+                    []() -> Some<CSGObject>
+                    {
+                        return Some<CSGObject>(new MockBaseClass);
+                    }
+                    ))))),
+    });
+
+    Manifest another_manifest("Mock library",
+    {
+        std::make_pair("mock_base_class", erase_type(
+            MetaClass<CSGObject>(erase_type(
+                std::function<Some<CSGObject>()>(
+                    []() -> Some<CSGObject>
+                    {
+                        return Some<CSGObject>(new MockBaseClass);
+                    }
+                    ))))),
+    });
+
+    EXPECT_EQ(manifest.description(), "Mock library");
+    EXPECT_EQ(another_manifest, manifest);
```

Comment:
you should try to use `gdb` or `lldb` and add a breakpoint here and step through it, to see why it doesn't actually gets true.